receiver/jaegerreceiver/internal/udpserver/udp_server.go

```diff
@@ -0,0 +1,103 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package udpserver
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+type PacketHandler interface {
```

Comment:
Q: is this code mostly copied from Jaeger?

Reply:
Why not? I'm not going to review brand new UDP code. The existing code is battle tested.